Documentation/gettingstarted/k8s-install-openshift-okd.rst

```diff
@@ -0,0 +1,202 @@
+======================
+
+1. Choose preferred cloud provider. This guide was tested in AWS, Azure & GCP.
+
+2. Read `OpenShift documentation <https://docs.okd.io/latest/welcome/index.html>`_ to find out about provider-specific prerequisites
+
+3. `Get OpenShift Installer <https://github.com/openshift/okd#getting-started>`_
+
+.. note:: It's highly recommended to read the docs, unless you have installed
+   OpenShift in the past. Here are a few notes that you may find useful.
+   
+   - with the AWS provider ``openshift-install`` will not work properly
+     when MFA credentials are stored in ``~/.aws/credentials``, traditional credentials are required
+   - with the Azure provider ``openshift-install`` will prompt for
+     credentials and store them in ``~/.azure/osServicePrincipal.json``, it
+     doesn't simply pickup ``az login`` credentials. It's recommended to
+     setup a dedicated service principal and use it
+   - with the GCP provider ``openshift-install`` will only work with a service
+     account key, which has to be set using ``GOOGLE_APPLICATION_CREDENTIALS``
+     environment variable (e.g. ``GOOGLE_APPLICATION_CREDENTIALS=service-account.json``)
+
+Create an OpenShift OKD Cluster
+===============================
+
+First, set cluster name:
+
+.. code:: bash
+
+   CLUSTER_NAME="cluster-1"
+
+Now, create configuration files:
+
+   NOTE: the sample output below is showing the AWS provider, but
+   it should work the same way with other providers
+
+.. code:: shell-session
+
+   $ openshift-install create install-config --dir "${CLUSTER_NAME}"
+   ? SSH Public Key ~/.ssh/id_rsa.pub
+   ? Platform aws
+   INFO Credentials loaded from default AWS environment variables 
+   ? Region eu-west-1
+   ? Base Domain openshift-test-1.cilium.rocks
+   ? Cluster Name cluster-1
+   ? Pull Secret [? for help] ******************************************************************************************************************************************************************************************
+
+And set ``networkType: Cilium``:
+
+.. code:: bash
+
+   sed -i 's/networkType:\ OVNKubernetes/networkType:\ Cilium/' "${CLUSTER_NAME}/install-config.yaml"
+
+Next, generate OpenShift manifests:
+
+.. code:: bash
+
+   openshift-install create manifests --dir "${CLUSTER_NAME}"
+
+Now, define ``cilium`` namespace:
+
+.. code:: bash
+
+   cat << EOF > "${CLUSTER_NAME}/manifests/cluster-network-03-cilium-namespace.yaml"
+   apiVersion: v1
+   kind: Namespace
+   metadata:
+     name: cilium
+     annotations:
+       # node selector is required to make cilium-operator run on control plane nodes
+       openshift.io/node-selector: ""
+     labels:
+       name: cilium
+       # run level sets priority for Cilium to be deployed prior to other components
+       openshift.io/run-level: "0"
+   EOF
+
+.. include:: k8s-install-download-release.rst
+
+Next, render Cilium manifest:
+
+.. parsed-literal::
+
+   helm template |CHART_RELEASE|  \\
```

Comment:
@seanmwinn are any of these keys going to change with #13259?

Reply:
Yes, all of them.

Reply:
I would be great if we could merge this to master first, and make an update separately. That would help with 1.8 backport.